modules/reactor/src/meshgenerators/PatternedHexMeshGenerator.C

```diff
@@ -502,6 +526,30 @@ PatternedHexMeshGenerator::generate()
 
   setMeshProperty("pattern_pitch_meta", _pattern_pitch);
 
+  // create a list of interface boundary ids for each input mesh
+  // NOTE: list of interface boundary ids is stored in mesh metadata
+  std::vector<std::set<boundary_id_type>> input_interface_boundary_ids;
+  input_interface_boundary_ids.resize(_input_names.size());
+  if (_use_interface_boundary_id_shift)
+  {
+    for (const auto i : make_range(_input_names.size()))
+    {
+      if (!hasMeshProperty<bool>("interface_boundaries", _input_names[i]))
+        mooseError("Metadata 'interface_boundaries' could not be found on the input mesh: ",
+                   _input_names[i]);
+      if (!getMeshProperty<bool>("interface_boundaries", _input_names[i]))
+        mooseError("Interface boundary ids were not constructed in the input mesh",
+                   _input_names[i]);
+      if (!hasMeshProperty<std::set<boundary_id_type>>("interface_boundary_ids", _input_names[i]))
+        mooseError("Metadata 'interface_boundary_ids' could not be found on the input mesh: ",
+                   _input_names[i]);
+    }
+  }
+  for (const auto i : make_range(_input_names.size()))
+    if (hasMeshProperty<std::set<boundary_id_type>>("interface_boundary_ids", _input_names[i]))
```

Comment:
I dont think we need to check anymore since there s a if (!) just before

Reply:
In case we do not apply id shifting (`_use_interface_boundary_id_shift == false`), we do not check `interface_boundary_ids` metadata. Thus, it is safe to check whether this metadata exists.

I think this one is better:
```c++
  for (const auto i : make_range(_input_names.size()))
    if (_use_interface_boundary_id_shift || hasMeshProperty<std::set<boundary_id_type>>("interface_boundary_ids", _input_names[i]))
      input_interface_boundary_ids[i] =
          getMeshProperty<std::set<boundary_id_type>>("interface_boundary_ids", _input_names[i]);
```

Reply:
yup you're right